File: lily/@lily/packages/violet/src/net/bodz/violet/schema/tran/AbstractTransportOrder.ts
import { int } from 'skel01-core/src/lang/basetype';
import CoMessage from 'lily-basic/src/net/bodz/lily/concrete/CoMessage';
import AbstractTransportOrderTypeInfo from './AbstractTransportOrderTypeInfo';
import Contact from 'lily-basic/src/net/bodz/lily/schema/contact/Contact';

export abstract class AbstractTransportOrder<Id> extends CoMessage<Id> {

    static readonly TYPE = AbstractTransportOrderTypeInfo.INSTANCE;

    src: Contact = new Contact();
    dst: Contact = new Contact();

    constructor(o: any) {
        super(o);
    }

}

export default AbstractTransportOrder;

File: lily/violet-types/src/main/java/net/bodz/violet/schema/tran/AbstractTransportOrder.java
package net.bodz.violet.schema.tran;

import net.bodz.lily.concrete.CoMessage;
import net.bodz.lily.meta.TypeParamType;
import net.bodz.lily.meta.TypeParameters;
import net.bodz.lily.schema.contact.Contact;

@TypeParameters({ TypeParamType.ID_TYPE })
public abstract class AbstractTransportOrder<Id>
        extends CoMessage<Id> {

    private static final long serialVersionUID = 1L;

    protected Contact src = new Contact();
    protected Contact dst = new Contact();

    public Contact getSrc() {
        return src;
    }

    public void setSrc(Contact src) {
        this.src = src;
    }

    public Contact getDst() {
        return dst;
    }

    public void setDst(Contact dst) {
        this.dst = dst;
    }

}
